scipy/stats/_qmc.py

```diff
@@ -87,6 +87,22 @@ def check_random_state(seed=None):
                          ' numpy.random.Generator instance')
 
 
+def _bounds_iv(l_bounds, u_bounds, d):
```

Comment:
I was going to suggest sharing implementation with the [`Bounds` class](https://github.com/scipy/scipy/blob/10451614e7a59028a3b2415d208636c448bea63e/scipy/optimize/_constraints.py#L207-L281), but needs here are a little different, and it's simple enough.

Reply:
I thought about this, but this is in `optimize` and we already have circular dependency issues with some optimize using LHS, etc. With the simplification there is not much more duplicate.

For latter, maybe we can think about moving `Bounds` in a general place.